File: src/types/gameContent.ts
export interface GameContentSummary {
  gameId: string
  versionLabel?: string
  versionDetail?: string
  dlcCount: number
  updateCount: number
  dlcNames: string[]
  hasBase: boolean
}

export interface HomeContentInventory {
  summaries: GameContentSummary[]
}

export type GameContentMap = Record<string, GameContentSummary>

export function toContentMap(inventory: HomeContentInventory): GameContentMap {
  return Object.fromEntries(inventory.summaries.map((summary) => [summary.gameId, summary]))
}

File: src-tauri/src/models/game_content.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct GameContentSummary {
    pub game_id: String,
    pub version_label: Option<String>,
    pub version_detail: Option<String>,
    pub dlc_count: u32,
    pub update_count: u32,
    pub dlc_names: Vec<String>,
    pub has_base: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct HomeContentInventory {
    pub summaries: Vec<GameContentSummary>,
}
